src/openssl.c

```diff
@@ -3249,17 +3249,49 @@ static BIO *getbio(lua_State *L) {
 	bio = lua_touserdata(L, -1);
 	lua_pop(L, 1);
 
-	BIO_reset(*bio);
-
 	return *bio;
+} /* getbioinstance() */
+
+
+static BIO *getbio(lua_State *L) {
+	BIO *bio = getbioinstance(L);
+	BIO_reset(bio);
+	return bio;
 } /* getbio() */
 
 
+static void pushbiostring(lua_State *L) {
```

Comment:
I don't love this helper function. I think I'd prefer to leave things how they were.